app/src/main/java/com/infomaniak/mail/data/MailData.kt

```diff
@@ -429,4 +473,18 @@ object MailData {
 
         return apiMessages.map { (apiMessage, _) -> apiMessage }
     }
+
+    /**
+     * Utils
+     */
+
+    private fun <T : BaseRealmObject> SharedFlow<ResultsChange<T>>.collectOnce(completion: (List<T>) -> Unit) {
+        var job: Job? = null
+        job = CoroutineScope(Dispatchers.IO).launch {
+            this@collectOnce.collect {
+                completion(it.list.toList())
+                job?.cancel()
+            }
+        }
+    }
 }
```

Comment:
No need to rewrite this method as similar methods already exist (`first()`, `firstOrNull()`, etc...).